sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java

```diff
@@ -397,6 +409,38 @@ private String idleConnectionValidation(Timestamps timestamps, Instant currentTi
         return errorMessage;
     }
 
+    private String isCancellationProneChannel(Timestamps timestamps, Instant currentTime, RntbdRequestManager requestManager, Channel channel) {
+        String errorMessage = StringUtils.EMPTY;
+
+        final Optional<RntbdContext> rntbdContext = requestManager.rntbdContext();
+
+        if (timestamps.cancellationCount() >= this.cancellationCountSinceLastReadThreshold) {
```

Comment:
Worth adding this comment as what we have on the transitTimeout validation: 

// Transit timeout can be a normal symptom under high CPU load.
 // When request timeout due to high CPU, 
// close the existing the connection and re-establish a new one will not help the issue but rather make it worse, return fast

Reply:
Fixed in next iteration.